lib/pid/src/PID.cpp

```diff
@@ -1,492 +1,120 @@
- * industry.
- *
- * This library is a port of Brett Beauregard's Arduino PID library:
- *
- *  http://www.arduino.cc/playground/Code/PIDLibrary
- *
- * The wikipedia article on PID controllers is a good place to start on
- * understanding how they work:
- *
- *  http://en.wikipedia.org/wiki/PID_controller
- *
- * For a clear and elegant explanation of how to implement and tune a
- * controller, the controlguru website by Douglas J. Cooper (who also happened
- * to be Brett's controls professor) is an excellent reference:
- *
- *  http://www.controlguru.com/
- *
- * Additions to original PID library:
- *
- * Autotuning capability, based on this library:
- * https://github.com/br3ttb/Arduino-PID-AutoTune-Library
- *
- * Associated blog post:
- * http://brettbeauregard.com/blog/2012/01/arduino-pid-autotune-library/
- *
- * General usage for autotuning:
- * 1. Create function that calculates required input and attach to a timer
- * interrupt so it can run in the background (or multithread it)
- * 2. Add actuator type to enum in PID class if it doesn't exist
- * 3. Add type cast for actuator type in setOutput() function if it doesn't exist
- * 4. Using PID object call setupAutoTune() passing it I/O pointers and actuator type
- * 5. Call autoTune() function, choose whether you want PI or PID control. Also can
- * pass a autoTuneConfig struct for autotune params (if nothing is passed, default
- * will be used).
- * 6. Highly recommend monitoring autotune loop, try to change outputStart and oStep
- * such that the output oscillates evenly about the setpoint
- * 7. Choose whether or not to use autoTune parameters, can set using
- * setAutoTuneParams() function
- *
- * General note: Don't fully rely on autoTune params, use this as a starting point
- * for finding the right PID tuning params and test if they work well or not
- */
-
-/**
- * Includes
- */
-
 #include "PID.h"
 
-PID::PID(float Kc, float tauI, float tauD, std::chrono::duration<float> interval) {
-  usingFeedForward = false;
-  inAuto           = false;
-
-  // Default the limits to the full range of I/O: 3.3V
-  // Make sure to set these to more appropriate limits for
-  // your application.
-  setInputLimits(0.0, 3.3);
-  setOutputLimits(0.0, 3.3);
-
-  tSample_ = interval;
+#include <algorithm>
+#include <mutex>
 
-  setTunings(Kc, tauI, tauD);
+using namespace PID;
 
-  setPoint_             = 0.0;
-  processVariable_      = 0.0;
-  prevProcessVariable_  = 0.0;
-  controllerOutput_     = 0.0;
-  prevControllerOutput_ = 0.0;
+PID::PID::PID(float proportionalGain, float intregralGain, float derivativeGain, float lowerBound, float upperBound,
+              float deadzone, bool antiKickback)
+    : m_PGain(proportionalGain),
+      m_IGain(intregralGain),
+      m_DGain(derivativeGain),
+      m_lowerBound(lowerBound),
+      m_upperBound(upperBound),
+      m_deadzone(deadzone),
+      m_antiKickback(antiKickback) {}
```

Comment:
I don't see the need for this constructor. I would rather force people to always pass in a config struct.